confdgnmi/src/confd_gnmi_client.py

```diff
@@ -226,38 +239,43 @@ def parse_args(args):
               datatype, subscription_mode, poll_interval, poll_count,
               read_count)
 
-    encoding = gnmi_pb2.Encoding.BYTES
+    encoding = dict(BYTES=gnmi_pb2.Encoding.BYTES,
+                    JSON=gnmi_pb2.Encoding.JSON,
+                    JSON_IETF=gnmi_pb2.Encoding.JSON_IETF)[opt.encoding]
     subscription_list = ConfDgNMIClient.make_subscription_list(
         prefix, paths, subscription_mode)
 
-    client = ConfDgNMIClient(HOST, PORT)
-    if opt.operation == "capabilities":
-        supported_models = client.get_capabilities()
-        print("Capabilities - supported models:")
-        for m in supported_models:
-            print("name:{} organization:{} version: {}".format(m.name,
-                                                               m.organization,
-                                                               m.version))
-    elif opt.operation == "subscribe":
-        print("Starting subscription ....")
-        client.subscribe(subscription_list,
-                         read_fun=ConfDgNMIClient.read_subscribe_responses,
-                         poll_interval=poll_interval, poll_count=poll_count,
-                         read_count=read_count)
-        print(".... subscription done")
-    elif opt.operation == "get":
-        notification = client.get(prefix, paths, datatype, encoding)
-        print("Get - Notifications:")
-        for n in notification:
-            ConfDgNMIClient.print_notification(n)
-    elif opt.operation == "set":
-        if len(paths) != len(vals):
-            log.warning("len(paths) != len(vals); %i != %i", len(paths),
-                        len(vals))
-            print(
-                "Number of paths (--path) must be the same as number of vals (--val)!")
-        else:
-            response = client.set(prefix, list(zip(paths, vals)))
+    with closing(ConfDgNMIClient(HOST, PORT)) as client:
+        if opt.operation == "capabilities":
+            supported_models = client.get_capabilities()
+            print("Capabilities - supported models:")
+            for m in supported_models:
+                print("name:{} organization:{} version: {}".format(m.name,
+                                                                   m.organization,
+                                                                   m.version))
+        elif opt.operation == "subscribe":
+            print("Starting subscription ....")
+            client.subscribe(subscription_list,
+                             read_fun=ConfDgNMIClient.read_subscribe_responses,
+                             poll_interval=poll_interval, poll_count=poll_count,
+                             read_count=read_count)
+            print(".... subscription done")
+        elif opt.operation == "get":
+            notification = client.get(prefix, paths, datatype, encoding)
+            print("Get - Notifications:")
+            for n in notification:
+                ConfDgNMIClient.print_notification(n)
```

Comment:
Pretty print the output should be added in the future.

Reply:
Yes, logging is definitely preferable. There are more issues with prints/logging in general (note this was already part of the original code, it just got shifted due to code structure change) - created a new issue #94.